### PR TITLE
chore dev guide cleanup

### PR DESCRIPTION
**Pull request by alexcg1**
- Renamed docs to fit naming scheme "<verb>ing" instead of "<verb>"
- Removed extraneous references to Jina (makes quicker to scan)
- Made titles more descriptive, focus on what user wants to do (e.g. "Ranking Results" instead of "Rankers")
- Moved a few things to more logical sections

- chore(dev-guide): clean up titles more, move executors up like in 2.0 readme
- chore(dev-guides): tidy up, better naming

### All Submissions:

* [x] Have you followed the [documentation style guide](https://github.com/jina-ai/docs/tree/master/page_templates/style_guide.md)? 

### New Feature Submissions:

1. [x] Have you added your new file to a table of content index? 
2. [ ] Did you use one of the templates in the [page_template](https://github.com/jina-ai/docs/tree/master/page_templates) folder?
3. [x] Is your title in Title Case? 
4. [ ] Are your section headers in sentence case?

### CODE REVIEW

**Comment by alexcg1** [comment]
Note: I didn't touch content of any files except the headings. So any other issues were there already. So don't bring them up here - let's save them for another PR ;)

File: chapters/flow/index.md [hunk 1]
```diff
@@ -1,4 +1,4 @@
-## Compose Your Workflow with the Flow API
+## Composing a Search Workflow with the Flow API
```
Comment (CatStark) [inline chapters/flow/index.md]: This should be sentence case

File: chapters/install/ide-speedup.md [hunk 2]
```diff
@@ -1,4 +1,4 @@
-# Speed-up Jina app development in PyCharm and VSCode
+# Speeding Up Development in PyCharm and VSCode
```
Comment (CatStark) [inline chapters/install/ide-speedup.md:1]: This should be Title Case

File: chapters/simple_exec.md [hunk 3]
```diff
@@ -1,4 +1,4 @@
-# Built-in Simple Executors and Reserved `uses` in Jina
+# Built-in Simple Executors and Reserved `uses`
```
Comment (CatStark) [inline chapters/simple_exec.md:1]: Also Title Case please  :)

**Review by CatStark — DISMISSED** [review]

File: chapters/install/ide-speedup.md (hunk 2 above)
Comment (alexcg1) [reply chapters/install/ide-speedup.md:1]: It already is

https://titlecase.com/

Screenshot from 2021-05-17 09-54-21

File: chapters/flow/index.md (hunk 1 above)
Comment (alexcg1) [reply chapters/flow/index.md]: This is more of a case of the heading being the wrong level. It should be `#`, not `##`, so case should remain as title. I'll fix the heading level bug though

File: chapters/simple_exec.md (hunk 3 above)
Comment (alexcg1) [reply chapters/simple_exec.md:1]: Again, it already is according to titlecase.com :)

File: chapters/mime-type/index.rst [hunk 4]
```diff
@@ -1,4 +1,4 @@
-MIME Types in Jina
+MIME Types
```
Comment (FionnD) [inline chapters/mime-type/index.rst:1]: I don't understand the difference between this page and Primitive Data Types. Additionally should this not all fall under input/ouput?

**Review by FionnD — CHANGES_REQUESTED** [review]
one question

File: chapters/mime-type/index.rst (hunk 4 above)
Comment (alexcg1) [reply chapters/mime-type/index.rst:1]: Let's fix that issue later. I don't want to rewrite or merge any docs right now. Just organize and standardize what we've got.

**Review by FionnD — APPROVED** [review]

**Review by CatStark — APPROVED** [review]
LGTM!

**Comment by nan-wang** [comment]
hmm, it seems quite a few links are broken. https://github.com/jina-ai/docs/pull/269/checks?check_run_id=2608096484

**Comment by nan-wang** [comment]
I'll merge this PR for the time being.

**Pull request merged**